docs/topics/lombok.md

```diff
@@ -0,0 +1,137 @@
+[//]: # (title: Lombok compiler plugin)
+
+> The Lombok compiler plugin is [Experimental](components-stability.md). 
+> It may be dropped or changed at any time. Use it only for evaluation purposes. 
+> We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-7112).
+>
+{type="warning"}
+
+The Lombok compiler plugin allows the generation and use of Java's Lombok declarations within the same mixed Java/Kotlin module.
+If you generate these declarations in a Java module and use them in another Kotlin module, 
+then you don't need to use this plugin.
+
+[Lombok](https://projectlombok.org/) generates Java code itself, not the Kotlin Lombok compiler plugin. 
+Thus, you still need to configure Lombok as usual, when using this plugin. 
+Learn more about [how to make the plugin seeing Lombok's config](#defining-the-place-of-lombok-config).
+
+## Support of annotations
+
+See which annotations the plugin supports and the
+[current state of Lombok compiler plugin](https://github.com/JetBrains/kotlin/blob/master/plugins/lombok/lombok-compiler-plugin/README.md).
+
+Currently, we don't have plans on supporting the @Builder annotation. However, we can consider this if you vote
+for [`@Builder` in YouTrack](https://youtrack.jetbrains.com/issue/KT-46959).
+
+> Kotlin compiler ignores Lombok annotations if you use them in Kotlin code.
+>
+{type="note"}
+
+## Gradle
+
+### Main settings
+
+Apply the `kotlin-plugin-lombok` Gradle plugin in the `build.gradle.kts` file:
+
+```kotlin
+plugins {
+    id ("org.jetbrains.kotlin.plugin.lombok") version "%kotlinVersion%"
+}
+```
+
+Set up the Lombok itself as the plugin:
+
+```kotlin
+plugins {
+    id ("io.freefair.lombok") version "5.3.0"
+}
+```
+
+See the [test project with examples of Lombok compiler plugin usage](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_gradle/nokapt).
+
+### Defining the place of lombok.config
```

Comment:
Actually it's "specifying". But maybe a simpler version will work. Smth like "Using the Lombok configuration file".